lib/dead_end/who_dis_syntax_error.rb

```diff
@@ -61,6 +61,9 @@ def on_parse_error(msg)
       when /unexpected .* expecting ['`]?(?<unmatched_symbol>[^']*)/
         @unmatched_symbol = $1.to_sym if $1
         @error_symbol = :unmatched_syntax
+      when /unexpected '(?<unmatched_symbol>.*)'/
+        @unmatched_symbol = $1.to_sym if $1
+        @error_symbol = :unexpected_syntax
```

Comment:
Introducing this to differentiate between these two cases:

```
def soap_request
  ]
end

➜  dead_end git:(main) ✗ ruby -wc soaprequest_deadend.rb
soaprequest_deadend.rb:2: syntax error, unexpected ']'
```

And:

```
def soap_request
  [
end

➜  dead_end git:(main) ✗ ruby -wc soaprequest_deadend.rb
soaprequest_deadend.rb:3: syntax error, unexpected `end', expecting ']'
```

Reply:
I think we can simplify the logic in the banner class by normalizing both cases here.

i.e. something like:

```
when /unexpected .* expecting ['`]?(?<unmatched_symbol>[^']*)/
  invert the symbol and save
when /unexpected '(?<unmatched_symbol>.*)'/ 
  don't invert symbol, and save
```